test/fixtures/spell-check/success0.md

```diff
@@ -2,27 +2,39 @@ This file should not generate any spell-check errors.
 
 ## Node.js
 
-[Node.js](https://foo.com) - Node.js
+[Node.js](https://nodejs.com) - Node.js
```

Comment:
In fact, it is a node with `type=link`, but spell-check rule only checks `type=text`. Those test cases gave you a false-positive impression that your changes worked well.

Reply:
Thanks, I'll look into these thoughts.

Reply:
I think this is okay because that node is type=link will have a child text node that `spell-check` will run on.

Reply:
@transitive-bullshit Yes, but the child text node is the link text, "Node.js". And we want to lint the link itself, "https://nodejs.com". See [AST Explorer Example](https://astexplorer.net/#/gist/34d0fc96f76fab709fe7dbdaad4ea512/799526e96bca89f3478eb442f11194753696a5b2).

Reply:
Yep, you're 100% right.